src/vma/dev/qp_mgr.cpp

```diff
@@ -536,15 +554,21 @@ void qp_mgr::mlx5_send(vma_ibv_send_wr *p_send_wqe)
 	 * implementations may use move-string-buffer assembler instructions,
 	 * which do not guarantee order of copying.
 	 */
-	COPY_64B_NT(dst, m_sq_hot_wqe);
+	mlx5_bf_copy((unsigned long long *)((uintptr_t)m_sq_bf_reg + m_sq_bf_offset),
+		(unsigned long long *)m_sq_hot_wqe);
+
 	m_sq_bf_offset ^= m_sq_bf_buf_size;
 
+	m_sq_wqe_idx_to_wrid[m_sq_hot_wqe_index] = (uintptr_t)p_send_wqe->wr_id;
+
+	++m_sq_wqe_counter;
```

Comment:
In the original code m_sq_wqe_counter was incremented earlier. Why do you delay its increment to this position?